src/mon/OSDMonitor.cc

```diff
@@ -963,10 +963,12 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
   dout(20) << "mon_stretch_cluster_recovery_ratio: " << cct->_conf.get_val<double>("mon_stretch_cluster_recovery_ratio") << dendl;
 	if (prev_num_up_osd < osdmap.num_up_osd &&
 	    (osdmap.num_up_osd / (double)osdmap.num_osd) >
-	    cct->_conf.get_val<double>("mon_stretch_cluster_recovery_ratio")) {
+	    cct->_conf.get_val<double>("mon_stretch_cluster_recovery_ratio") &&
+      mon.dead_mon_buckets.size() == 0) {
```

Comment:
Hmm. This definitely works for our current 2-site setup, but we'll need to adjust it if we start supporting 3-site (and other count) stretch clusters with the explicit stretch mechanisms. That was something I was considering (and trying to keep easy) when writing the other code.
So I'd rather do something that won't require changing when we hit that point, but I don't have a good simple solution, so looks good.

Reply:
make sense. Thank you